### DISCUSSION TITLE
Issue Starting ARM Container -> Config Folder Permissions not Setting?

### DISCUSSION THREAD

**Discussion by blairmitch73** [automatic-ripping-machine/automatic-ripping-machine#1057] [Q&A]
Installing Arm Via Dockers.
Short Version is Getting this Error in the logs
```
[OK]: ARM UID and GID set correctly, ARM has access to '/home/arm' using 1002:1002
Removing any link between music and Music
Checking ownership of /etc/arm/config
---------------------------------------------
[ERROR]: ARM does not have permissions to /etc/arm/config using 1002:1002
Check your user permissions and restart ARM. Folder permissions--> 0:0
---------------------------------------------
*** /etc/my_init.d/arm_user_files_setup.sh failed with status 1

```
Config files is as follows

#!/bin/bash
docker run -d \
    -p "8081:8081" \
    -e ARM_UID="1002" \
    -e ARM_GID="1002" \
    -v "/home/arm:/home/arm" \
    -v "/yacht/AppData/Logs/arm:/home/arm/logs" \
    -v "/yacht/Media/Movies:/home/arm/media" \
    -v "/yacht/AppData/Config/arm:/etc/arm/config" \
    --device="/dev/sr0:/dev/sr0" \
    --device="/dev/sr1:/dev/sr1" \
    --device="/dev/sr2:/dev/sr2" \
    --device="/dev/sr3:/dev/sr3" \
    --privileged \
    --restart "always" \
    --name "arm-rippers" \
    --cpuset-cpus='4,5' \
    automaticrippingmachine/automatic-ripping-machine:latest

Thoughts?

**Comment by blairmitch73** [comment]
Looking at the file setup .sh 
I notice that they check for ownership BEFORE the file is created.

##### Setup ARM-specific config files if not found
# Check ownership of the ARM config folder
check_folder_ownership "/etc/arm/config"

mkdir -p /etc/arm/config

**Reply by microtechno9000** [reply]
The script checks for permissions, and stops ARM from starting because it will fail if the ownership is wrong.

looks like you are mounting the volumes using samba or NFS? suggest checking those mounted systems permissions are set correctly 
Looks like NFS is set with UID and GID being squashed to 0, hence the 0:0

**Reply by simeongarner** [reply]
I am having the same error.  I am mounting with CIFS, all the permissions are set correctly.  Just wondering if a solution, even temporary, has been found?

**Reply by microtechno9000** [reply]
The script previously forced a change ownership to the user and group per the docker startfile. This was removed as its a bad idea to force a change ownership and instead check for ownership, as over mounted network shares it can cause issues.

Suggest changing the owner to arm:arm, or what ever uid and gid you have in your start script. Then restart the container.

**Reply by Greybeard-1963** [reply]
I am currently experiencing the same issue with a clean install of  Ubuntu 20.04.6 LTS. I have tried creating a folder /etc/arm/config with the permissions of arm:arm, even going as far as setting the permissions to 777 to no avail.

**Reply by microtechno9000** [reply]
Does the UID and GID in the docker file align? 

@Greybeard-1963 can you post the output of your log that is failing

**Reply by Greybeard-1963** [reply]
Thanks for getting back to me so swiftly, @microtechno9000. 

Yes they do, it fails at the point where it checks the permissions for /etc/arm/config and just goes round in an infinite loop not allowing shell access to the container. I tried running the startup script with the ID's defined and with the lines removed but no difference. It also seems to affect 2.6.69, but 2.6.53 start fine and the web UI comes up and the CD is polled. At this point I managed to attach a terminal and realised it was the file structure within the container image it couldn't access, rather my local copy of Ubuntu.

Don't know if this related in any way, but I tried the latest release at the time (2.6.69) with Ubuntu 22.04 letting the install script create the users and group itself and running the script via sudo. Under that scenario, it failed at the /home/arm section.
 
I'm not at my PC ATM, but I'll post the configs etc. later.

**Reply by microtechno9000** [reply]
Thanks for the info and testing various versions

Once you get the logs, that will help so I can see what's wrong

**Reply by Greybeard-1963** [reply]
Sorry for taking so long getting back to you, was testing various scenarios.

Bottom line, 2.6.71 seems OK, 2.6.69 seems to be broken. Things got really confusing last night as I was pulling LATEST while you rolled from 2.6.69 => 2.6.70.

Some interesting observations, if you run 2.6.71 from  start_arm_container.sh the container loads and runs fine. However, if you perform a docker run IMAGE_ID it falls over. Conversely, it runs fine as docker run IMAGE_ID but if you run 2.6.69 from  start_arm_container.sh, the container loops forever.

Test and logs etc. below. Many, many thanks for getting me up and running. 

<details><summary>Details</summary>
<p>
=====================================================================================
docker version test
=====================================================================================

2.6.71    FAIL docker run image - PASS running start_arm_container.sh
2.6.70    FAIL docker run image - script not tested
2.6.69    PASS docker run image - FAIL running start_arm_container.sh
2.6.67    PASS docker run image - (Included for completeness)
2.6.53    PASS docker run image - (Included for completeness)

=====================================================================================
2.6.71 Fail log (Using docker run image)
=====================================================================================

*** Running /etc/my_init.d/10_syslog-ng.init...
Mar  4 02:15:55 079a5fd00ccc syslog-ng[12]: syslog-ng starting up; version='3.25.1'
*** Running /etc/my_init.d/arm_user_files_setup.sh...
Updating arm group id 1000 to default (1000)...
usermod: no changes
Updating arm group id 1000 to default (1000)...
Adding arm user to 'render' group
Checking ownership of /home/arm
[OK]: ARM UID and GID set correctly, ARM has access to '/home/arm' using 1000:1000
Creating dir: /home/arm/media
Creating dir: /home/arm/media/completed
Creating dir: /home/arm/media/raw
Creating dir: /home/arm/media/movies
Creating dir: /home/arm/media/transcode
Creating dir: /home/arm/logs
Creating dir: /home/arm/logs/progress
Creating dir: /home/arm/db
Creating dir: /home/arm/music
Creating dir: /home/arm/.MakeMKV
Removing any link between music and Music
stat: cannot stat '/etc/arm/config': No such file or directory
stat: cannot stat '/etc/arm/config': No such file or directory
Checking ownership of /etc/arm/config
---------------------------------------------
[ERROR]: ARM does not have permissions to /etc/arm/config using 1000:1000
Check your user permissions and restart ARM. Folder permissions--> :
---------------------------------------------
*** /etc/my_init.d/arm_user_files_setup.sh failed with status 1

===================================================================================== 
working start_arm_container.sh 
===================================================================================== 

 #!/bin/bash                                                                                                                                                            
 docker run -d \
     -p "8080:8080" \
     -v "/home/arm/ARM:/home/arm" \
     -v "/home/arm/ARM/music:/home/arm/music" \
     -v "/home/arm/ARM/logs:/home/arm/logs" \
     -v "/home/arm/ARM/media:/home/arm/media" \
     -v "/home/arm/ARM/config:/etc/arm/config" \
     --device="/dev/sr0:/dev/sr0" \
     --privileged \
     --restart "always" \
     --name "arm-rippers" \
     --cpuset-cpus='0,1' \
     automaticrippingmachine/automatic-ripping-machine:2.6.71

=====================================================================================
Fail log 2.6.69 (docker run IMAGE_ID)
=====================================================================================

*** Running /etc/my_init.d/10_syslog-ng.init...
Mar  4 05:10:14 feeb90f79948 syslog-ng[12]: syslog-ng starting up; version='3.25.1'
*** Running /etc/my_init.d/arm_user_files_setup.sh...
Updating arm group id 1000 to default (1000)...
usermod: no changes
Updating arm group id 1000 to default (1000)...
Adding arm user to 'render' group
Creating dir: /home/arm/media/completed
Creating dir: /home/arm/media/raw
Creating dir: /home/arm/media/movies
Creating dir: /home/arm/media/transcode
Creating dir: /home/arm/logs/progress
Creating dir: /home/arm/db
Creating dir: /home/arm/.MakeMKV
Config not found! Creating config file: /etc/arm/config/arm.yaml
Config not found! Creating config file: /etc/arm/config/apprise.yaml
rm: cannot remove '/home/arm/music': Is a directory
*** /etc/my_init.d/arm_user_files_setup.sh failed with status 1

*** Killing all processes...
Mar  4 05:10:17 feeb90f79948 syslog-ng[12]: syslog-ng shutting down; version='3.25.1'
*** Running /etc/my_init.d/10_syslog-ng.init...
Mar  4 05:10:18 feeb90f79948 syslog-ng[12]: syslog-ng starting up; version='3.25.1'
*** Running /etc/my_init.d/arm_user_files_setup.sh...
Updating arm group id 1000 to default (1000)...
usermod: no changes
Updating arm group id 1000 to default (1000)...
Adding arm user to 'render' group
rm: cannot remove '/home/arm/music': Is a directory
*** /etc/my_init.d/arm_user_files_setup.sh failed with status 1

*** Killing all processes...
Mar  4 05:10:19 feeb90f79948 syslog-ng[12]: syslog-ng shutting down; version='3.25.1'
*** Running /etc/my_init.d/10_syslog-ng.init...
Mar  4 05:10:20 feeb90f79948 syslog-ng[12]: syslog-ng starting up; version='3.25.1'
*** Running /etc/my_init.d/arm_user_files_setup.sh...
Updating arm group id 1000 to default (1000)...
usermod: no changes
Updating arm group id 1000 to default (1000)...
Adding arm user to 'render' group
rm: cannot remove '/home/arm/music': Is a directory
*** /etc/my_init.d/arm_user_files_setup.sh failed with status 1

*** Killing all processes...
Mar  4 05:10:21 feeb90f79948 syslog-ng[12]: syslog-ng shutting down; version='3.25.1'
*** Running /etc/my_init.d/10_syslog-ng.init...
Mar  4 05:10:23 feeb90f79948 syslog-ng[12]: syslog-ng starting up; version='3.25.1'
*** Running /etc/my_init.d/arm_user_files_setup.sh...
Updating arm group id 1000 to default (1000)...
usermod: no changes
Updating arm group id 1000 to default (1000)...
Adding arm user to 'render' group
rm: cannot remove '/home/arm/music': Is a directory
*** /etc/my_init.d/arm_user_files_setup.sh failed with status 1

*** Killing all processes...
Mar  4 05:10:24 feeb90f79948 syslog-ng[12]: syslog-ng shutting down; version='3.25.1'
*** Running /etc/my_init.d/10_syslog-ng.init...
Mar  4 05:10:25 feeb90f79948 syslog-ng[12]: syslog-ng starting up; version='3.25.1'
*** Running /etc/my_init.d/arm_user_files_setup.sh...
Updating arm group id 1000 to default (1000)...
usermod: no changes
Updating arm group id 1000 to default (1000)...
Adding arm user to 'render' group
rm: cannot remove '/home/arm/music': Is a directory
*** /etc/my_init.d/arm_user_files_setup.sh failed with status 1

*** Killing all processes...
Mar  4 05:10:26 feeb90f79948 syslog-ng[12]: syslog-ng shutting down; version='3.25.1'
*** Running /etc/my_init.d/10_syslog-ng.init...
Mar  4 05:10:29 feeb90f79948 syslog-ng[12]: syslog-ng starting up; version='3.25.1'
*** Running /etc/my_init.d/arm_user_files_setup.sh...
Updating arm group id 1000 to default (1000)...
usermod: no changes
Updating arm group id 1000 to default (1000)...
Adding arm user to 'render' group
rm: cannot remove '/home/arm/music': Is a directory
*** /etc/my_init.d/arm_user_files_setup.sh failed with status 1

*** Killing all processes...
Mar  4 05:10:30 feeb90f79948 syslog-ng[12]: syslog-ng shutting down; version='3.25.1'

===================================================================================== 
Truncated docker container list (Running start_arm_container.sh - loops forever)
=====================================================================================

CONTAINER ID   COMMAND           CREATED         STATUS                          PORTS     
feeb90f79948   "/sbin/my_init"   6 minutes ago   Restarting (1) 14 seconds ago   

</p>
</details>

**Reply by 3n1gmus** [reply]
I had this issue, I was able to get past it by changing the ARM_UID and ARM_GID to the UID and GID manually to the respective IDs.

-e ARM_UID="1001" \
-e ARM_GID="1001" \
